src/main/java/Duke/Command/Command.java

```diff
@@ -0,0 +1,294 @@
+package Duke.Command;
+
+import Duke.Exceptions.DukeException;
+import Duke.Exceptions.IndexNotInListException;
+import Duke.Exceptions.NoDescriptionException;
+import Duke.Exceptions.WrongInputException;
+
+import Duke.Storage.Storage;
+import Duke.Tasks.Task;
+import Duke.Tasks.TaskList;
+import Duke.Tasks.Todo;
+import Duke.Tasks.Deadline;
+import Duke.Tasks.Event;
+import Duke.Ui.Ui;
+import java.util.ArrayList;
+
+/**
+ * The Command class is responsible for directing the user's interaction with Duke.
+ */
+public abstract class Command {
+    private static boolean exit;
+
+    /**
+     * Executes the specified action that is expected of the command.
+     *
+     * @param tasklist Takes in a tasklist to update the list of tasks
+     * @param ui Takes in an instance of the ui class to show details to the user
+     * @param storage Takes in an instance of the storage to update and renew the txt files.
+     * @throws DukeException If an error caused by Duke's limitation occurs
+     */
+    public abstract void execute(TaskList tasklist, Ui ui, Storage storage) throws DukeException;
+
+    public static class ListCommand extends Command {
+
+        @Override
+        public void execute(TaskList tasklist, Ui ui, Storage storage) {
+            ui.showTasklist(tasklist);
+        }
+
+    }
+
+    public static class ClearCommand extends Command {
+
+        /**
+         * Executes the clearing of the task list.
+         *
+         * @param tasklist Takes in a tasklist to update the list of tasks
+         * @param ui Takes in an instance of the ui class to show details to the user
+         * @param storage Takes in an instance of the storage to update and renew the txt files.
+         */
+        @Override
+        public void execute(TaskList tasklist, Ui ui, Storage storage) {
+            tasklist.deleteAll();
+            ui.showTasklistDeleted();
+            storage.updateTxtFile(tasklist);
+        }
+    }
+
+    public static class FindCommand extends Command {
+        String input;
+
+        public FindCommand(String input) {
+            this.input = input;
+        }
+
+        @Override
+        public void execute(TaskList tasklist, Ui ui, Storage storage) throws NoDescriptionException {
+            if (input.trim().toLowerCase().equals("find")) {
+                String message = "Oi, what you want me find?!";
+                throw new NoDescriptionException(message);
+            } else {
+                String keyword = input.split(" ")[1].trim();
+                ArrayList<Task> searches = tasklist.find(keyword);
+                ui.showSearches(searches);
+            }
+        }
+    }
+
+    public static class DoneCommand extends Command {
+        String input;
```

Comment:
Perhaps you would like to define an access modifier here? But it's just for clarity, this should default to package-private.